lua/nvim-lsp-ts-utils/import-all.lua

```diff
@@ -177,30 +177,50 @@ local response_handler_factory = function(callback)
     end
 end
 
+-- if a module has more than one import edit returns true
+local should_reorder = function(edits)
+    local modules = {}
+    for _, edit in ipairs(edits) do
+        if edit.newText then
+            for module in string.gmatch(edit.newText, "import.*from (.+)[;\n]?") do
+                local import = module:gsub("\n", "")
+                if modules[import] then
+                    return true
+                end
+                modules[import] = true
+                break
```

Comment:
@jose-elias-alvarez honestly I don't know the difference between match and gmatch. Would you mind explaining further the idea? I don't often work in Lua and still don't know its APIs well.

Reply:
They're really similar except `match` returns the actual matches while `gmatch` returns an iterator. In this case, since we only expect one match at most, we can do something like this, which IMO is a bit clearer:

```lua
local should_reorder = function(edits)
    local modules = {}
    for _, edit in ipairs(edits) do
        if edit.newText then
            local module = string.match(edit.newText, "import.*from (.+)[;\n]?")
            -- this shouldn't be necessary but we can be safe
            if not module then
                return
            end

            local import = module:gsub("\n", "")
            if modules[import] then
                return true
            end

            modules[import] = true
        end
    end
    return false
end
```

Reply:
@jose-elias-alvarez gotcha! I'll change this code and add the config option `always_organize_imports` (default to true).